### ISSUE TITLE
Corrupted text in TEdit when printing long output to TEXTSTREAM

### ISSUE THREAD

**Issue by pamoroso** [Interlisp/medley#1697]
**Describe the bug**
Suppose you create a `TEXTSTREAM` and associate it with a new TEdit window. If you print at least several dozen lines of output to the `TEXTSTREAM`, the text eventually ends up corrupted in the TEdit window.

**To Reproduce**
1. sign into your Interlisp Online account
2. under `Initial Exec`, select `Interlisp`
3. click `Run Medley`
4. evaluate `(SETQ TS (OPENTEXTSTREAM))`
5. evaluate `(TEDIT TS)`
6. evaluate `(FOR I FROM 1 TO 200 DO (PRINTOUT TS "Line no. " I T))`

**Expected behavior**
All the text appears in the TEdit window with no corruption or alterations.

**Screenshots**
The TEdit window:

![textstream1](https://github.com/Interlisp/medley/assets/10342479/0effdd08-3161-4b44-b145-9b5fe56cc455)

The full desktop:

![textstream2](https://github.com/Interlisp/medley/assets/10342479/c2907af2-1730-46d6-a290-9255399721c3)

**Context (please complete the following information):**
 - OS: chromeOS Stable
 - OS Version: 124.0.6367.95
 - Host arch: x86_64 (ASUS Chromebox 3)
 - Maiko version: N/A
 - IL:MAKESYSDATE: 22-Apr-2024 02:03:43

**Additional context**
Sometimes the text corruption appears when printing only a couple dozen lines of text. The issue occurs also when printing to the `TEXTSTREAM` from Common Lisp, for example with `format`.

**Comment by pamoroso** [comment]
Does `OPENTEXTSTREAM` require the first argument, i.e. a stream? Can it be omitted?

**Comment by pamoroso** [comment]
Passing a stream to `OPENTEXTSTREAM` like this makes no difference as the corrupted text issue still occurs:

```lisp
(SETQ S (OPENSTREAM '{NODIRCORE} 'BOTH))

(SETQ TS (OPENTEXTSTREAM S))
```

**Comment by rmkaplan** [comment]
Thanks for finding this so clearly, I’ve seen this sporadically but haven’t been able to track it down.

It was an order of execution error in an early attempt to speed up the case of appending to the end of a stream. When it was updating the window as well as just updating the text, it was not properly reasserting the file position when the line formatter was trying to determine whether it was running off the bottom of the window.

Appending to a text in an open window is still much to slow.  One issue is that the inherited algorithms for formatting and in particular displaying at the end of a stream are proportional to the length of the preceding characters on the line, and maybe even in the paragraph.  A substantial speed up for appending would, among other things, make it feasible for TRACE to print to a text stream so that the trace output would be scrollable.

**Comment by pamoroso** [comment]
I noticed the slowness and was indeed wondering. `TEDIT.INSERT` is much faster and has no corruption issues.

**Comment by pamoroso** [comment]
Never mind, I'm getting text corruption issues or an error that something is not a file descriptor also when adding text with `TEDIT.INSERT`, deleting all of the text with `(TEDIT.DELETE TS 1 (TEDIT.NCHARS TS))`, adding some different text, and deleting everything again with `TEDIT.DELETE`. I'm trying to reproduce the issue.

**Comment by pamoroso** [comment]
I managed to trigger the error `ARG NOT LINEDESCRIPTOR` when calling `TEDIT.DELETE` (it wasn't about file descriptors, I misremembered). Here's the session:

![tedit-delete](https://github.com/Interlisp/medley/assets/10342479/7306aa72-2a38-4b57-8862-9a7186abc98e)

**Comment by rmkaplan** [comment]
Thanks, got it.

It wasn’t the insert, it was in the screen update after deleting everything.

**Comment by pamoroso** [comment]
Resolved by #1857.

**Comment by pamoroso** [comment]
Resolved by #1857.
